repos/system_upgrade/common/actors/peseventsscanner/libraries/pes_events_scanner.py

```diff
@@ -0,0 +1,462 @@
+    relevant_releases_match_list = [
+        '> {0}'.format(api.current_actor().configuration.version.source),
+        '<= {0}'.format(api.current_actor().configuration.version.target)
+    ]
+    releases = {event.to_release for event in events}
+    releases = [r for r in releases if version.matches_version(relevant_releases_match_list, '{}.{}'.format(*r))]
+    return sorted(releases)
+
+
+def _get_enabled_modules():
+    enabled_modules_msgs = api.consume(EnabledModules)
+    enabled_modules_msg = next(enabled_modules_msgs, None)
+    if list(enabled_modules_msgs):
+        api.current_logger().warning('Unexpectedly received more than one EnabledModules message.')
+    if not enabled_modules_msg:
+        raise StopActorExecutionError('Cannot parse PES data properly due to missing list of enabled modules',
+                                      details={'Problem': 'Did not receive a message with enabled module '
+                                                          'streams (EnabledModules)'})
+    return enabled_modules_msg.modules
+
+
+def compute_pkg_changes_between_consequent_releases(source_installed_pkgs, events, release, seen_pkgs):
+    # Start with the installed packages and modify the set according to release events
+    target_pkgs = set(source_installed_pkgs)
+
+    release_events = [e for e in events if e.to_release == release]
+
+    for event in release_events:
+        # PRESENCE events have a different semantics than the other events - they add a package to a target state
+        # only if it had been seen (installed) during the course of the overall target packages
+        if event.action == Action.PRESENT:
+            for pkg in event.in_pkgs:
+                if pkg in seen_pkgs:
+                    if pkg in target_pkgs:
+                        # Remove the package with the old repository, add the one with the new one
+                        target_pkgs.remove(pkg)
+                    target_pkgs.add(pkg)
+        elif event.action == Action.DEPRECATED:
+            if event.in_pkgs.intersection(source_installed_pkgs):
+                # Remove packages with old repositories add packages with the new one
+                target_pkgs = target_pkgs.difference(event.in_pkgs)
+                target_pkgs = target_pkgs.union(event.in_pkgs)
+        else:
+            # All other packages have the same semantics - they remove their in_pkgs from the system with given
+            # from_release and add out_pkgs to the system matching to_release
+            are_all_in_pkgs_present = all(in_pkg in source_installed_pkgs for in_pkg in event.in_pkgs)
+            is_any_in_pkg_present = any(in_pkg in source_installed_pkgs for in_pkg in event.in_pkgs)
+
+            # For MERGE to be relevant it is sufficient for only one of its in_pkgs to be installed
+            if are_all_in_pkgs_present or (event.action == Action.MERGED and is_any_in_pkg_present):
+                # In pkgs are present, event can be applied
```

Comment:
```suggestion
                # If pkgs are present, event can be applied
```

Reply:
That's not an If ... then do ..., it's a statement saying that they are present

Reply:
@vinzenz is right, its a statement about what does the condition above the comment imply when the branch is taken.